### PR TITLE
fix: Clip canvas when rendering Game

### PR DESCRIPTION
**Pull request by st-pasha**
# Description

Ensure that game canvas is clipped, allowing it to be used in situations where the game was not fullscreen. See https://github.com/bluefireteam/dashbook/issues/68 for an example of this bug.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [ ] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ ] I have updated/added relevant examples in `examples`.

## Breaking Change

- [x] No, this is *not* a breaking change.

## Related Issues

Fixes https://github.com/bluefireteam/dashbook/issues/68

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: packages/flame/lib/src/game/game_render_box.dart [hunk 1]
```diff
@@ -67,6 +67,7 @@ class GameRenderBox extends RenderBox with WidgetsBindingObserver {
   void paint(PaintingContext context, Offset offset) {
     context.canvas.save();
     context.canvas.translate(offset.dx, offset.dy);
+    context.canvas.clipRect(Rect.fromLTWH(0, 0, size.width, size.height));
```
Comment (spydon) [inline packages/flame/lib/src/game/game_render_box.dart:70]: You don't want to create a new `Rect` here on every tick.

**Review by spydon — COMMENTED** [review]
I wonder how expensive this operation is

File: packages/flame/lib/src/game/game_render_box.dart (hunk 1 above)
Comment (st-pasha) [reply packages/flame/lib/src/game/game_render_box.dart]: ok

**Comment by wolfenrain** [comment]
> I wonder how expensive this operation is

I believe we had this done on purpose, @erickzanardo can you shed some light on this? We had a discussion about it few months ago

**Comment by erickzanardo** [comment]
> > I wonder how expensive this operation is
> 
> I believe we had this done on purpose, @erickzanardo can you shed some light on this? We had a discussion about it few months ago

Yeah, we don't clip the canvas because that is a Flutter "standard", things are not clipped by default, you have to clip things yourself when you want this. The `Clip` widget helps a lot on this, so I don't think Flame should clip itself.

**Comment by st-pasha** [comment]
@erickzanardo I thought you wanted to add clipping to Flame: https://github.com/bluefireteam/dashbook/issues/68#issuecomment-953677947

**Comment by spydon** [comment]
> @erickzanardo I thought you wanted to add clipping to Flame: https://github.com/bluefireteam/dashbook/issues/68#issuecomment-953677947

From that comment Erick says that it should be in the examples app, not in the core code. So that the `GameWidget` is wrapped in a `Clip` widget in examples/.

**Pull request closed without merging**